### PR TITLE
Feature - Blurry Dialog Background

### PR DESCRIPTION
**Pull request by corvinsz**
I recently came across @Keboo 's video ([C#/Blazor - Trying out MudBlazor on an app](https://youtu.be/tb-q596Kv8k?t=3341)) about MudBlazor, where they have implemented a Dialog that blurs the background if opened.
This is my aproach of implementing the above mechanism in MDIX.

## API proposal
I added 2 new dependency properties to the `DialogHost.cs`:
```xaml
<materialDesign:DialogHost ApplyBlurBackground="True"
                           BlurRadius="8"
                           ...>
</materialDesign:DialogHost>
```
| Property            | Type     | Default value |
| ------              | ------   | ------        |
| ApplyBlurBackground | `Bool`   | `False`       |
| BlurRadius          | `double` | 16            |

I essentially just added the 2 new dependency properties above, and some logic to apply a `BlurEffect` on the `DialogHost`s `ContentPresenter`.
The logic is contained in the newly added converter `DialogBackgroundBlurConverter`.

| ApplyBlurBackground | BlurRadius    | Default Style                                                                                       | MaterialDesignEmbeddedDialogHost                                                                          |
| --------            | --------      | -------                                                                                             | -------                                                                                                   |
| True                | 16 (Default)  | image | image |
| True                | 64            | image | image |

~~I don't know why GitHub shows the images in the column "Default Style" in a smaller format, all images are the same exact size :-(~~ (Apparently it's just the preview of the markdown that is not display it correctly)

### CODE REVIEW

File: src/MaterialDesignThemes.Wpf/Converters/DialogBackgroundBlurConverter.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
+    {
+        if (values is [bool isOpen, bool applyBlurBackground, double blurRadius]
+            && isOpen
+            && applyBlurBackground)
+        {
+            return new BlurEffect() { Radius = blurRadius };
+        }
+
+        return null;
```
[the remark this answers is not in the document]
Comment (nicolaihenriksen) [reply src/MaterialDesignThemes.Wpf/Converters/DialogBackgroundBlurConverter.cs:23]: @corvinsz It's the analyzers yelling at you here. Basically you need to add a ? to the return type, the `values` parameter, and the `parameter` parameter to indicate nullability.

[the remark this answers is not in the document]
Comment (corvinsz) [reply src/MaterialDesignThemes.Wpf/Converters/DialogBackgroundBlurConverter.cs:23]: Thank you for clarifying @nicolaihenriksen 
I'm still doing alot of .NET Framework 4.x work due to legacy projects and I always forget about the nullability in .NET 5+

File: src/MaterialDesignThemes.Wpf/Converters/DialogBackgroundBlurConverter.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    public object[] ConvertBack(object value, Type[] targetTypes, object parameter, CultureInfo culture) => throw new NotImplementedException();
```
Comment (Keboo) [inline src/MaterialDesignThemes.Wpf/Converters/DialogBackgroundBlurConverter.cs:25]:
```suggestion
    public object?[]? ConvertBack(object? value, Type[] targetTypes, object? parameter, CultureInfo culture) => throw new NotImplementedException();
```

**Review by Keboo — COMMENTED** [review]
Very nicely done. One small comment then I think we can merge this.

**Review by Keboo — APPROVED** [review]
Awesome job, thank you!

**Pull request merged**